network/irc/irc.go

```diff
@@ -596,6 +614,29 @@ func (bot *ircBot) act(theLine *line.Line) {
 		return
 	}
 
+	isAskingForSasl := theLine.User == "" && strings.HasSuffix(theLine.Raw, " CAP * ACK :sasl")
```

Comment:
Yep, for instance servers may or may not include that final `:` before the `sasl` because it's entirely optional, similar with uppercasing the command and all.